app/controllers/concerns/recaptcha_concern.rb

```diff
@@ -11,6 +11,14 @@ module RecaptchaConcern
     'https://recaptcha.google.com/recaptcha/',
   ].freeze
 
+  def add_recaptcha_resource_hints
+    response.headers['Link'] = [
+      response.headers['Link'],
+      '<https://www.google.com>;rel=preconnect',
+      '<https://www.gstatic.com>;rel=preconnect;crossorigin',
```

Comment:
would it be worth trying to dynamically build these from `RECAPTCHA_SCRIPT_SRC` or is it just easier to hardcode?

Reply:
I don't know that there's any guarantees that these map one-to-one with the CSP headers, strictly-speaking.